large-message-core/src/main/java/com/bakdata/kafka/CompressionType.java

```diff
@@ -0,0 +1,142 @@
+package com.bakdata.kafka;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.nio.ByteBuffer;
+
+import org.apache.kafka.common.errors.SerializationException;
+import org.apache.kafka.common.record.RecordBatch;
+import org.apache.kafka.common.utils.BufferSupplier;
+import org.apache.kafka.common.utils.ByteBufferOutputStream;
+
+import software.amazon.awssdk.utils.IoUtils;
+
+public enum CompressionType {
+    NONE(0, "none") {
+        @Override
+        public byte[] compress(byte[] bytes) {
+            return bytes;
+        }
+
+        @Override
+        public byte[] decompress(byte[] bytes) {
+            return bytes;
+        }
+    },
+
+    GZIP(1, "gzip") {
+        @Override
+        public byte[] compress(byte[] bytes) {
+            return CompressionType.compress(org.apache.kafka.common.record.CompressionType.GZIP, bytes);
+        }
+
+        @Override
+        public byte[] decompress(byte[] bytes) {
+            return CompressionType.decompress(org.apache.kafka.common.record.CompressionType.GZIP, bytes);
+        }
+    },
+
+    SNAPPY(2, "snappy") {
+        @Override
+        public byte[] compress(byte[] bytes) {
+            return CompressionType.compress(org.apache.kafka.common.record.CompressionType.SNAPPY, bytes);
+        }
+
+        @Override
+        public byte[] decompress(byte[] bytes) {
+            return CompressionType.decompress(org.apache.kafka.common.record.CompressionType.SNAPPY, bytes);
+        }
+    },
+
+    LZ4(3, "lz4") {
+        @Override
+        public byte[] compress(byte[] bytes) {
+            return CompressionType.compress(org.apache.kafka.common.record.CompressionType.LZ4, bytes);
+        }
+
+        @Override
+        public byte[] decompress(byte[] bytes) {
+            return CompressionType.decompress(org.apache.kafka.common.record.CompressionType.LZ4, bytes);
+        }
+    },
+
+    ZSTD(4, "zstd") {
+        @Override
+        public byte[] compress(byte[] bytes) {
+            return CompressionType.compress(org.apache.kafka.common.record.CompressionType.ZSTD, bytes);
+        }
+
+        @Override
+        public byte[] decompress(byte[] bytes) {
+            return CompressionType.decompress(org.apache.kafka.common.record.CompressionType.ZSTD, bytes);
+        }
+    };
+
+    private static byte[] compress(org.apache.kafka.common.record.CompressionType compressionType, byte[] bytes) {
+        ByteBufferOutputStream outStream = new ByteBufferOutputStream(bytes.length);
+        try (OutputStream stream = compressionType.wrapForOutput(outStream, RecordBatch.MAGIC_VALUE_V2)) {
```

Comment:
Could you please explain what the magic value is for?

Reply:
from some code reading I know the name of the argument is `messageVersion`. for most compression types the argument is ignored.
for lz4 there's a check to see if it's `MAGIC_VALUE_V0`, in which case it sets `useBrokenFlagDescriptorChecksum` to true for the compression it does there.
so I just picked the latest `MAGIC_VALUE_VX` to pass in